cmake/compilerFlags.cmake

```diff
@@ -20,17 +20,24 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
         endif()
     endif()
 
-    if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
+    if (COMPILER_IS_GCC)
+        add_compile_options(-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS)

```

Comment:
These flags are glibc features and iirc are also provided by clang. Also `FORTIFY_SOURCE` requires to compile with `-O2`, so this will give you warnings on non-release builds.

Reply:
Good to know all these things, I will make the corresponding changes :+1: 

I actually observed some warnings when using this with apple-clang:
https://travis-ci.org/piponazo/exiv2/jobs/564657957